### PR TITLE
Check storage layout consistency in PRs

### PR DESCRIPTION
**Pull request by Amxx**
Fixes LIB-674

Adds a new step in the CI checks that will compare the storage layout between the PR and the reference branch.

### CODE REVIEW

**Comment by frangio** [comment]
The layout comparison script looks very good. But the build info artifacts are about 100MB each, which is a lot of overhead to be downloading/uploading on every run. I believe we should instead store the layout objects as the artifacts, i.e. the result of `extractLayouts`.

**Comment by Amxx** [comment]
Good point, I have not thought about it. I will split the extract/compare into 2 scripts, and we can save the intermediary representation.

**Comment by Amxx** [comment]
Done. From my tests the layout in json format are ~35k

**Comment by Amxx** [comment]
Does this require a changeset ?

File: .github/actions/storage-layout/action.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        gh run download ${RUN_ID} --repo ${{ github.repository }} -n layout
+      env:
+        GITHUB_TOKEN: ${{ inputs.token }}
+      shell: bash
+      continue-on-error: true
+      id: reference
+    - name: Compare layouts
+      if: steps.reference.outcome == 'success' && github.event_name == 'pull_request'
+      run: |
+        node scripts/checks/layout-rebuild-and-compare.js --head $${{ inputs.layout }} --ref ${{ inputs.ref_layout }} >> $GITHUB_STEP_SUMMARY
```
Comment (frangio) [inline .github/actions/storage-layout/action.yml:43]: Is this correct? Shouldn't it be using `compare-layouts.js`? And as far as I can tell `layout-rebuild-and-compare.js` should be removed?

Comment (Amxx) [reply .github/actions/storage-layout/action.yml:43]: yes

**Review by frangio — APPROVED** [review]
Looks good!

**Pull request merged**